nixos/dwm-status:  use structured RFC42 style settings (#416212)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2511.section.md

- The `services.siproxd` module has been removed as `siproxd` is unmaintained and broken with libosip 5.x.

> - The `dwm-status` module now uses `settings` instead of plain-text `extraConfig` and generates the config file completely from Nix attrs (ensuring [RFC0042](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md) compliancy).

Reviewer: What about this wording instead? Personally I'd also omit the "which is RFC0042 compliant" part as I don't think the end user needs to know about it. For removing the technical details like "plain-text" and "Nix attrs", I think the user should look at the options documentation for reference on how to migrate, an example for it would be nice (see next comment)
```suggestion
- `services.dwm-status.extraConfig` was removed. It has been replaced by `services.dwm-status.settings` which is [RFC0042](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md) compliant.
```
